C++ API parity: LeakyReLU

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27309 C++ API parity: MultiheadAttention
* #27382 C++ API parity: Linear
* #27060 C++ API parity: LogSigmoid
* **#27059 C++ API parity: LeakyReLU**

Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

pbelevich committed:
C++ API parity: LeakyReLU

[ghstack-poisoned]

yf225 commented on torch/csrc/api/include/torch/nn/functional/activation.h:
```diff
@@ -29,6 +29,14 @@ inline Tensor hardtanh(const Tensor& input, const HardtanhOptions& options) {
 
+inline Tensor leaky_relu(const Tensor& input, const LeakyReLUOptions& options) {
+  if (options.inplace()) {
+    return torch::leaky_relu_(const_cast<Tensor&>(input), options.negative_slope());
```
same here for changing `input`'s type to `Tensor&`

yf225 commented on torch/csrc/api/include/torch/nn/options/activation.h:
```diff
@@ -42,5 +42,18 @@ struct HardtanhOptions {
+  // Controls the angle of the negative slope. Default: 1e-2
+  TORCH_ARG(double, negative_slope) = 1e-2;
+
+  // can optionally do the operation in-place. Default: False
```
nit: need `///` instead of `//` (similarly for other PRs)

yf225 commented on torch/csrc/api/src/nn/modules/activation.cpp:
```diff
@@ -54,5 +54,22 @@ void HardtanhImpl::pretty_print(std::ostream& stream) const {
+void LeakyReLUImpl::pretty_print(std::ostream& stream) const {
+  stream << std::boolalpha
+         << "torch::nn::LeakyReLU(negative_slope=" << options.negative_slope()
+         << ", inplace=" << options.inplace() << ")";
```
same here, whether to print `inplace` should be conditional.

pbelevich committed:
Update on "C++ API parity: LeakyReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"




[ghstack-poisoned]

yf225 reviewed (approved):
Thanks @pbelevich !

pbelevich committed:
Update on "C++ API parity: LeakyReLU"


Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"


Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"


Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"


Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"


Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"


Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"


Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LeakyReLU"


Differential Revision: [D17682407](https://our.internmc.facebook.com/intern/diff/D17682407)

[ghstack-poisoned]
